### PR TITLE
Opacity node

### PR DESCRIPTION
**Pull request by theflyingzamboni**
Separated opacity from the Blend Node and made it a standalone adjustment node.

Migration is intended to only create an Opacity Node from a Blend Node when Opacity != 100 to reduce clutter. Seems to be covering various possible conditions (e.g. opacity value, whether an edge is connected to an image input, iterator parent), but please try to break to make sure.

Closes joeyballentine/chaiNNer#425

### CODE REVIEW

File: backend/src/nodes/image_util_nodes.py [hunk 1]
```diff
@@ -24,26 +24,10 @@ def __init__(self):
         """Constructor"""
         super().__init__()
         self.description = """Blends overlay image onto base image using 
-            specified mode and opacities."""
+            specified mode."""
         self.inputs = [
             ImageInput("Base Layer"),
-            SliderInput(
```
Comment (RunDevelopment) [inline backend/src/nodes/image_util_nodes.py:30]: Is it really a good idea to remove them? I understand you can achieve the same effect by using the new opacity node, but I still think it's usefull to have them here.

Also, the opacity sliders make it very clear that an RGBA (well, BGRA actually) is returned. Without the sliders, users might expect that e.g. the diff of 2 1-channel images to be a 1-channel image.

Comment (theflyingzamboni) [reply backend/src/nodes/image_util_nodes.py:30]: I've actually toyed with the idea that it should output an image with c = max(base, overlay). I wasn't sure whether it was actually worth it, but I might do that, at least if the node doesn't have its own sliders. As for whether blend should have its own opacity sliders, I think that might warrant some discussion on discord.

File: backend/src/nodes/image_adj_nodes.py [hunk 2]
```diff
@@ -253,3 +253,44 @@ def run(
...
+        self.category = IMAGE_ADJUSTMENT
+        self.name = "Opacity"
+        self.icon = "MdOutlineOpacity"
+        self.sub = "Adjustments"
+
+    def run(self, img: np.ndarray, opacity: float) -> np.ndarray:
+        """Apply opacity adjustment to alpha channel"""
+
+        if opacity == 100:
+            return img
```
Comment (RunDevelopment) [inline backend/src/nodes/image_adj_nodes.py:287]: I think this is bad.

The return value of a node should be consistent. This `if` also makes it impossible to statically type the node correctly. I'd have to type the number of output channels as `4 | InputImage.channels` which is just not as useful of just `4`.

Comment (RunDevelopment) [reply backend/src/nodes/image_adj_nodes.py:287]: Please make the condition `opacity == 100 and c == 4`.

Comment (theflyingzamboni) [reply backend/src/nodes/image_adj_nodes.py:287]: Hmm. Was trying to avoid unnecessary conversions, but I also haven't looked into your typing system.

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**